src/aics-image-viewer/components/CellViewerCanvasWrapper/index.jsx

```diff
@@ -1,74 +1,72 @@
-import React from 'react';
-import { View3d } from '@aics/volume-viewer';
+import React from "react";
+import { View3d } from "@aics/volume-viewer";
 
-import { Icon } from 'antd';
+import { Icon } from "antd";
 
-import viewMode from '../../shared/enums/viewMode';
+import viewMode from "../../shared/enums/viewMode";
 
-import AxisClipSliders from '../AxisClipSliders';
-import { BottomPanel } from '../BottomPanel';
-import './styles.css';
+import AxisClipSliders from "../AxisClipSliders";
+import { BottomPanel } from "../BottomPanel";
+import "./styles.css";
 
 const ViewMode = viewMode.mainMapping;
 
 export default class ViewerWrapper extends React.Component {
- constructor(props) {
-   super(props);
-   this.view3dviewerRef = React.createRef();
-   this.getActiveAxis = this.getActiveAxis.bind(this);
-   this.setAxisClip = this.setAxisClip.bind(this);
-   this.renderOverlay = this.renderOverlay.bind(this);
-   this.renderClipSliders = this.renderClipSliders.bind(this);
- }
-
- componentDidMount() {
-   if (!this.view3D) {
-     this.view3D = new View3d(this.view3dviewerRef.current);
-     this.props.onView3DCreated(this.view3D);
-   }
- }
-
- componentWillReceiveProps(newProps) {
-   if (this.view3D && this.props.mode && this.props.mode !== newProps.mode) {
-     this.view3D.setCameraMode(viewMode.VIEW_MODE_ENUM_TO_LABEL_MAP.get(newProps.mode));
-   }
-   if (this.view3D && this.props.autorotate !== newProps.autorotate) {
-     this.view3D.setAutoRotate(newProps.autorotate);
-   }
-
-   this.view3D.resize();
+  constructor(props) {
+    super(props);
+    this.view3dviewerRef = React.createRef();
+    this.getActiveAxis = this.getActiveAxis.bind(this);
+    this.setAxisClip = this.setAxisClip.bind(this);
+    this.renderOverlay = this.renderOverlay.bind(this);
+    this.renderClipSliders = this.renderClipSliders.bind(this);
+  }
+
+  componentDidMount() {
+    if (!this.view3D) {
+      this.view3D = new View3d(this.view3dviewerRef.current);
+      this.props.onView3DCreated(this.view3D);
+      this.view3D.setAutoRotate(this.props.autorotate);
+    }
+  }
+
+  componentDidUpdate(prevProps, _prevState) {
+    if (this.view3D && prevProps.mode && prevProps.mode !== this.props.mode) {
+      this.view3D.setCameraMode(viewMode.VIEW_MODE_ENUM_TO_LABEL_MAP.get(this.props.mode));
```

Comment:
Worth talking generally about what component should be responsible for propagating which changes to `view3d` in this PR, since I chose not to move `backgroundColor` into this component in #41, piggybacked off of `autorotate` already being in this component here, and having more clarity on this will be useful in #32.

I'm a bit concerned about the added complexity of letting this component be responsible for calls to `view3d` that don't require access to `image`. In order to keep this component fully responsible for calling `setAutoRotate`, I put the initial call in `componentDidMount` here (potentially a non-obvious place for it). The other method that this component appears to manage, `setCameraMode`, is not initially called here, but in the base `App` component, and is currently being called twice every time `ControlPanel` calls the update handler: once by `App` in `handleChangeToImage`, then a second time here when this component receives new props. I think it makes some sense to move responsibility for calling `view3d` methods to handlers in `App` to mitigate similar double-call issues in the future and reduce complexity a bit, even if it doesn't make the most semantic sense.

Reply:
I see your point.  I can see it both ways.  Imagine if this React component "CellViewerCanvasWrapper" were coming from the `volume-viewer` library.    Shouldn't it have props including `backgroundColor` and `autoRotate`?
On the other hand, if the CellViewerCanvasWrapper component exposes the view3d, then it doesn't need any other means of accessing the volume-viewer functionality.
I suppose right now I would slightly lean to agreeing with you -- that the view3d is already being cached up at the app level and all calls made in it from that layer.

Reply:
I definitely see tradeoffs for doing it both ways - it's very nice at the app level to be able to just call a state change and have `CellViewerCanvasWrapper` propagate that to `view3d` on its own. I will add as well that I spent a while debugging some strange behavior after modifying `App` to turn off `autorotate` in this manner when switching to 2d view modes, which turned out to be due to the fact that `view3d` maintains separate control states for 2d and 3d modes, and `componentDidUpdate` only gets called to sync that change after `App` has already changed modes and `view3d` has already swapped control states. So while convenient, I'd still say updating `view3d` in component lifecycles isn't quite worth the footguns.